### PR TITLE
Story 52 And Merchant Dashboard

### LINKED ISSUE #62: User Story 52, Merchant sees an order show page
```
As a merchant
When I visit an order show page from my dashboard
I see the customer's name and address
I only the items in the order that are being purchased from my inventory
I do not see any items in the order being purchased from other merchants
For each item, I see the following information:
- the name of the item, which is a link to my item's show page
- a small thumbnail of the item
- my price for the item
- the quantity the user wants to purchase
```

DO THESE STORIES FIRST:
- User Story 35, Merchant Dashboard displays Orders

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #63

child of #125

depends on #82

### LINKED ISSUE #82: User Story 35, Merchant Dashboard displays Orders
```
As a merchant
When I visit my dashboard ("/dashboard")
If any users have pending orders containing items I sell
Then I see a list of these orders.
Each order listed includes the following information:
- the ID of the order, which is a link to the order show page ("/dashboard/orders/15")
- the date the order was made
- the total quantity of my items in the order
- the total value of my items for that order
```

DO THESE STORIES FIRST:
- User Story 34, Merchant Dashboard Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #84

child of #125

depends on #83

### LINKED ISSUE #83: User Story 34, Merchant Dashboard Show Page
```
As a merchant user
When I visit my dashboard ("/dashboard")
I see my profile data, but cannot edit it
```

DO THESE STORIES FIRST:
- User Story 7, User Registration
- User Story 4, Merchant Navigation

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #84

child of #125

depends on #116

depends on #120

### PR DESCRIPTION
**Pull request by cnodland**
closes #62 
closes #83 
closes #82 

Also added Merchant Dashboard functionality

### CODE REVIEW

File: app/controllers/merchants/orders_controller.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
 class Merchants::OrdersController < Merchants::BaseController
   def index
+    merchant = User.find(current_user.id)
+    @orders = Order.find_by_merchant(merchant)
+  end
+
+  def show
+    merchant = User.find(current_user.id)
+    @order = Order.find(params[:id])
+    @items = Item.find_by_order(@order, merchant)
```
Comment (wipegup) [inline app/controllers/merchants/orders_controller.rb]: Maybe `find_by_order` could be an `Order` instance method that takes the merchant as an argument?

```
@items = @order.merchants_items(merchant)
```

Where:
```
class Order < Application Record
  #Code
  def merchants_items(merchant)
    items.where(user:merchant)
  end
end
```

My thinking is that we have already "pulled" out the order information, so we might as well use that in finding the items

**Review by wipegup — APPROVED** [review]
Looks good

**Review by wipegup — CHANGES_REQUESTED** [review]
I will be making changes on the factories to un-break some tests

**Review by wipegup — APPROVED** [review]

**Review by whois-jon-peterson — APPROVED** [review]

**Review by Vjp888 — APPROVED** [review]
